lib/virsh_autotest_base.pm

```diff
@@ -0,0 +1,126 @@
+# VIRSH TEST MODULE BASE PACKAGE
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+#
+# Summary: This is the base package for virsh test modules, for example,
+# tests/virtualization/xen/hotplugging.pm
+# tests/virt_autotest/virsh_internal_snapshot.pm
+# tests/virt_autotest/virsh_external_snapshot.pm
+# and etc.
+# Maintainer: Wayne Chen <wchen@suse.com>
+
+package virsh_autotest_base;
+
+use base "consoletest";
+use strict;
+use warnings;
+use POSIX 'strftime';
+use File::Basename;
+use Data::Dumper;
+use XML::Writer;
+use IO::File;
+use testapi;
+use utils;
+use virt_utils;
+use xen;
+
+sub run_test {
+    die('Please override this subroutine in children modules to run actual tests.');
```

Comment:
I am not sure if this approach is correct.

Reply:
At least I tried with automation run.

Reply:
In my opinion, if you want to run something at the beginning of every test module, just make a function out of it, put it into the base module and use it everywhere you need.

Reply:
Hello @asmorodskyi, what do you think about this? I am not sure.

Reply:
'sub run' in this file is a typical such case, this function run_test is an exception that the child classes should implement their real test, which should be different.